### PR TITLE
Add vertex feature store integration

### PR DESCRIPTION
**Pull request by eliasecchig**
Authors: @lspataroG @eliasecchig 

Add VertexAI Langchain integration

[Example usage](https://github.com/eliasecchig/fs_genai/blob/main/vertex_feature_store.ipynb)

### CODE REVIEW

File: libs/vertexai/pyproject.toml [hunk 1]
```diff
@@ -17,6 +17,11 @@ google-cloud-aiplatform = "^1.47.0"
 google-cloud-storage = "^2.14.0"
 # optional dependencies
 anthropic = { extras = ["vertexai"], version = ">=0.23.0,<1", optional = true }
+pandas = { version = ">= 1.0.0", optional = true }
+google-cloud-bigquery-storage = { version = ">=2.6.0,<3",  optional = true }
+pyarrow = { version = ">= 6.0.1", optional = true }
+langchain-community = { version = ">=0.0.28", optional = true }
```
Comment (lkuligin) [inline libs/vertexai/pyproject.toml:23]: we shouldn't depend on community

File: libs/vertexai/pyproject.toml [hunk 2]
```diff
@@ -17,6 +17,11 @@ google-cloud-aiplatform = "^1.47.0"
...
+db-dtypes = { version = "^1.2.0", optional = true }
```
Comment (lkuligin) [inline libs/vertexai/pyproject.toml:24]: should this be in the test group?

File: libs/vertexai/tests/integration_tests/vectorstores/feature_store/test_feature_store_bf_executor.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        },
+        {
+            "kind": "treat",
+        },
+        {
+            "kind": "fruit",
+        },
+    ]
+
+    @pytest.mark.skipif(os.getenv("RUN_SKIPPED") != "1", reason="CI/CD not ready.")
```
Comment (lkuligin) [inline libs/vertexai/tests/integration_tests/vectorstores/feature_store/test_feature_store_bf_executor.py:86]: just mark these tests as `extended`, please, and add missing infra to our internal test project

File: libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py [hunk 4]
```diff
@@ -0,0 +1,959 @@
...
+
+    def init_feature_store(self):
+        self.online_store_name = self.online_store_name or self._env_config.dataset_name
+        self.view_name = self.view_name or self._env_config.table_name
+        self.location = self.location or self._env_config.location
+        vertexai.init(project=self._env_config.project_id, location=self.location)
+
+        api_endpoint = f"{self._env_config.location}-aiplatform.googleapis.com"
+        self._admin_client = FeatureOnlineStoreAdminServiceClient(
+            client_options={"api_endpoint": api_endpoint}
```
Comment (lkuligin) [inline libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py:590]: please, pass a valid `client_info` for telemetry

File: libs/vertexai/pyproject.toml (hunk 2 above)
Comment (eliasecchig) [reply libs/vertexai/pyproject.toml:24]: no it's required for pandas - BQ interaction

File: libs/vertexai/pyproject.toml (hunk 1 above)
Comment (eliasecchig) [reply libs/vertexai/pyproject.toml:23]: Without that, we would need to bring this function our side though 

https://github.com/langchain-ai/langchain/blob/37cfc003107ea800953be912f2eebfbf069c9587/libs/community/langchain_community/vectorstores/utils.py#L23

Wdyt?

Comment (eliasecchig) [reply libs/vertexai/pyproject.toml]: took out MMR search for the time being

File: libs/vertexai/tests/integration_tests/vectorstores/feature_store/test_feature_store_bf_executor.py (hunk 3 above)
Comment (eliasecchig) [reply libs/vertexai/tests/integration_tests/vectorstores/feature_store/test_feature_store_bf_executor.py]: done

File: libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py (hunk 4 above)
Comment (eliasecchig) [reply libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py]: done thanks!

Comment (eliasecchig) [reply libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py]: done, thanks!

**Comment by gmogr** [comment]
In integration tests could you please add a check for already created datasets in BQ or add a unique postfix to the dataset name. 
tests/integration_tests/vectorstores/feature_store/test_feature_store_fs_executor.py are failing if BQ dataset is already created.

**Comment by eliasecchig** [comment]
@gmogr can you link an example where tests failed? 
If dataset already exists, the class should not complain about it, see [here](https://github.com/langchain-ai/langchain-google/pull/250/files#diff-88c67c3a7d16b1bc7e3665aab66bfc0eda199d83545f01bb2ec67851183ec828R211)

File: libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py [hunk 5]
```diff
@@ -0,0 +1,956 @@
...
+        )
+        self._online_store = self._create_online_store()
+        self._search_client = self._get_search_client()
+        self._feature_view = self._get_feature_view()
+
+    def _get_search_client(self) -> FeatureOnlineStoreServiceClient:
+        gca_resource = self._online_store.gca_resource
+        endpoint = gca_resource.dedicated_serving_endpoint.public_endpoint_domain_name
+        return FeatureOnlineStoreServiceClient(
+            client_options={"api_endpoint": endpoint}
```
Comment (lkuligin) [inline libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py:595]: it looks like telemetry is missing here too

File: libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/feature_store.py [hunk 6]
```diff
@@ -0,0 +1,567 @@
...
+        except ModuleNotFoundError as e:
+            logger.error(e)
+            raise ImportError(
+                "Please, install feature store dependency group: "
+                "`pip install langchain-google-vertexai[featurestore]`"
+            )
+        self._bq_client = bigquery.Client(
+            project=self.project_id,
+            location=self.location,
+            credentials=self.credentials,
```
Comment (lkuligin) [inline libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/feature_store.py:103]: same here, client_info is not set

File: libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/feature_store.py [hunk 7]
```diff
@@ -0,0 +1,567 @@
...
+            text content.
+        text_embedding_field (str): The field name in the Feature Store that stores
+            the text embeddings.
+        doc_id_field (str): The field name in the Feature Store that stores the
+            document IDs.
+        credentials (Optional[Any]): Optional credentials for Google Cloud
+            authentication.
+
+    To use, you need the following packages installed:
+        google-cloud-bigquery
```
Comment (lkuligin) [inline libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/feature_store.py:66]: can we hint a specific `langchain-google-vertexai` dep group instead, please?

File: libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py [hunk 8]
```diff
@@ -0,0 +1,956 @@
...
+        k: int = 5,
+        batch_size: Union[int, None] = None,
+    ) -> list[list[list[Any]]]:
+        raise NotImplementedError()
+
+    def set_env_config(self, env_config: EnvConfig):
+        self._env_config = env_config
+
+
+class BruteForceExecutor(BaseExecutor):
```
Comment (lkuligin) [inline libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py:79]: why do we need our own in-memory implementation if LC already has a few? can we just write an adapter that will load data into it?

**Pull request closed without merging**